sidechain/storage/src/storage.rs

```diff
@@ -97,6 +101,57 @@ impl<SignedBlock: SignedBlockT> SidechainStorage<SignedBlock> {
 		self.db.get(block_hash)
 	}
 
+	/// Get all blocks following (i.e. children of) a specified blocks
+	pub fn get_blocks_following(
+		&self,
+		block_hash: &BlockHash,
+		shard_identifier: &ShardIdentifierFor<SignedBlock>,
+	) -> Result<Vec<SignedBlock>> {
+		// Ensure we find the block in storage (otherwise we would return all blocks for a specific shard).
+		if self.get_block(block_hash)?.is_none() {
+			warn!("Could not find starting block in storage, returning empty vector");
+			return Ok(Vec::new())
+		}
+
+		// We get the latest block and then traverse the parents until we find our starting block.
+		let latest_block = match self.last_block_of_shard(shard_identifier) {
+			None =>
+				return Err(Error::LastBlockNotFound(
+					"Failed to find last block information".to_string(),
+				)),
+			Some(last_block_ref) => match self.get_block(&last_block_ref.hash)? {
+				None =>
+					return Err(Error::LastBlockNotFound(
+						"Failed to retrieve last block from storage".to_string(),
+					)),
+				Some(last_block) => last_block,
+			},
+		};
+
+		let mut current_block = latest_block;
+		let mut blocks_to_return = Vec::<SignedBlock>::new();
+		while &current_block.hash() != block_hash {
+			let parent_block_hash = current_block.block().parent_hash();
+
+			// Break in case we're at the genesis block
+			if parent_block_hash == BlockHash::default() {
+				break
+			}
+
+			blocks_to_return.push(current_block);
+
+			current_block = match self.get_block(&parent_block_hash)? {
+				None => return Err(Error::FailedToFindParentBlock),
+				Some(b) => b,
+			}
+		}
+
+		// Reverse because we iterate from newest to oldest, but result should be oldest first.
```

Comment:
sometimes you say "latest", "last" or "oldest". Is it the same? Then maybe unify?

Reply:
Hm - might be worth a thought, but what should be the consensus? Because both - first + last and oldest + newest are applicable in a blockchain.

And regarding your question - oldest = first, the newest would be the last block.